src/Workspaces/Core/Portable/DesignerAttribute/DesignerInfo.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.DesignerAttribute
+{
+    internal struct DesignerInfo
+    {
+        /// <summary>
+        /// Our current serialization format.  Increment whenever it changes so that we don't read
+        /// bogus data when reading older persisted data.
+        /// </summary>
+        private const string SerializationFormat = "1";
+
+        /// <summary>
+        /// The category specified in a <c>[DesignerCategory("...")]</c> attribute.
+        /// </summary>
+        public string? Category;
```

Comment:
>ublic string? Category; [](start = 9, length = 23)

Why mutable structs?

Reply:
because it's a jsonrpc serialization object.  making them immutable just adds more work and effort.  all i'm trying to say is: here are the two pieces of data returned by this rpc call.

Reply:
Are tuples not supported by the JSON serializer?

---
In reply to: [392395411](https://github.com/dotnet/roslyn/pull/42346#discussion_r392395411) [](ancestors = 392395411)

Reply:
Isn't this mixing two serializations? JSON RPC and persistent storage? That's why it's confusing. 
A mutable struct with just two mutable fields and nothing else would be ok. I'd also use a name that indicates its serialziation purpose 

---
In reply to: [392402394](https://github.com/dotnet/roslyn/pull/42346#discussion_r392402394) [](ancestors = 392402394,392395411)

Reply:
I don't really want to use tuples here.  it just means a ton of signatures where i have to spell out this tuple information, instead of having a nice name to refer to .

Reply:
> Isn't this mixing two serializations? JSON RPC and persistent storage? That's why it's confusing.

I removed the persistent stoarge code out of here.  It's now just json rpc.  I've added a comment explaining that.